doc/source/whatsnew/v1.4.0.rst

```diff
@@ -186,6 +186,38 @@ representation of :class:`DataFrame` objects (:issue:`4889`).
     df
     df.to_dict(orient='tight')
 
+.. _whatsnew_140.enhancements.shift:
+
+DataFrame.shift and Series.shift now accept an iterable for parameter ``'period'`` and new parameter ``'suffix'``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :meth:`DataFrame.shift` and :meth:`Series.shift` functions can take in an iterable, such as a list, for the period parameter. When an iterable is passed
+to either function it returns a :class:`DataFrame` object with all of the shifted rows or columns concatenated with one another.
+The function applies a shift designated by each element in the iterable. The resulting :class:`DataFrame` object's columns will retain the
```

Comment:
you don't need this sentence to the end. you can say that the ``suffix`` parameter controls the names but don't have to fully explain here. that's the point of the doc-string.